REGRESSION(269895@main):[WPE] 52 new API tests failures

cdumez opened the pull request:
#### 42d022005f544c9377c3227854cd4158dbdae361
<pre>
REGRESSION(269895@main):[WPE] 52 new API tests failures
<a href="https://bugs.webkit.org/show_bug.cgi?id=263974">https://bugs.webkit.org/show_bug.cgi?id=263974</a>

Reviewed by Michael Catanzaro.

Revert part of 269895@main which would cause the WebProcessProxy to outlive
its WebProcessPool in some cases, unnecessarily.

Also add a null check for the process pool is the WebProcessProxy destructor
before using it, for extra safety.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::isResponsive):
(WebKit::WebProcessProxy::isResponsiveWithLazyStop):

Canonical link: <a href="https://commits.webkit.org/270070@main">https://commits.webkit.org/270070@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d54a508f799b11277b25ae83a658acb3359fc5a0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24468 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/3018 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25729 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26595 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22506 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4660 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/450 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/22915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24712 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2092 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21136 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27181 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1816 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28262 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22305 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22385 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26055 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1757 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/81 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3041 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5857 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2210 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2132 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -337,7 +337,8 @@ WebProcessProxy::~WebProcessProxy()
 #if HAVE(DISPLAY_LINK)
-    processPool().displayLinks().stopDisplayLinks(m_displayLinkClient);
+    if (RefPtr processPool = processPoolIfExists())
+        processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
```
I notice `processPoolIfExists()` will log an error if the process pool doesn't exist, but this is an expected condition in the destructor, so an error is probably not appropriate, right? Alternative:

```suggestion
    if (RefPtr processPool = m_processPool)
        processPool->displayLinks().stopDisplayLinks(m_displayLinkClient);
```

Actually, I have no clue when it is appropriate to use `processPoolIfExists()`. Maybe it should just not log an error? I don't know.

cdumez replied:
Yes, the logging here would be unfortunate. I'll fix.

mcatanzaro reviewed (approved):
This change seems good.

cdumez committed:
REGRESSION(269895@main):[WPE] 52 new API tests failures
https://bugs.webkit.org/show_bug.cgi?id=263974

Reviewed by Michael Catanzaro.

Revert part of 269895@main which would cause the WebProcessProxy to outlive
its WebProcessPool in some cases, unnecessarily.

Also add a null check for the process pool is the WebProcessProxy destructor
before using it, for extra safety.

* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::WebProcessProxy::~WebProcessProxy):
(WebKit::WebProcessProxy::isResponsive):
(WebKit::WebProcessProxy::isResponsiveWithLazyStop):

Canonical link: https://commits.webkit.org/270070@main
